src/diagnostics/index.ts

```diff
@@ -0,0 +1,12 @@
+import { Disposable, languages } from "vscode";
+import { provideSingleton } from "../utils";
+
+@provideSingleton(DbtPowerUserDiagnostics)
+export class DbtPowerUserDiagnostics implements Disposable {
+  public readonly rebuildManifestDiagnostics =
+    languages.createDiagnosticCollection("dbt");
```

Comment:
nit: the collection name "dbt" seems too generic . Not sure what the scope of this is and if it may lead to global collisions. If yes, then let's make it specific to our extension like "dpu-dbt"

Reply:
This is not how it works. the list is just an instance and need to be managed as such. the name just shows up as the source.

Clear would only impact that instance.

<img width="2262" alt="image" src="https://github.com/innoverio/vscode-dbt-power-user/assets/9265503/29920d09-805e-49a5-bc23-99b32d0f17c7">